src/kernels/jupyter/launcher/serverUriStorage.ts

```diff
@@ -101,37 +103,45 @@ export class JupyterServerUriStorage implements IJupyterServerUriStorage {
         );
     }
     public async getSavedUriList(): Promise<{ uri: string; time: number; displayName?: string | undefined }[]> {
-        // List is in the global memento, URIs are in encrypted storage
-        const indexes = this.globalMemento.get<{ index: number; time: number }[]>(Settings.JupyterServerUriList);
-        if (indexes && indexes.length > 0) {
-            // Pull out the \r separated URI list (\r is an invalid URI character)
-            const blob = await this.encryptedStorage.retrieve(
-                Settings.JupyterServerRemoteLaunchService,
-                Settings.JupyterServerRemoteLaunchUriListKey
-            );
-            if (blob) {
-                // Make sure same length
-                const split = blob.split(Settings.JupyterServerRemoteLaunchUriSeparator);
-                const result = [];
-                for (let i = 0; i < split.length && i < indexes.length; i += 1) {
-                    // Split out the display name and the URI (they were combined because display name may have secret tokens in it too)
-                    const uriAndDisplayName = split[i].split(Settings.JupyterServerRemoteLaunchNameSeparator);
-                    const uri = uriAndDisplayName[0];
+        if (this.lastSavedList) {
+            return this.lastSavedList;
+        }
+        const promise = async () => {
```

Comment:
(tip for reviewers)
This PR reads much better with whitespaces hidden: https://github.com/microsoft/vscode-jupyter/pull/10017/files?w=1